### DISCUSSION TITLE
Auto-generator doesn't choose most current release as base diff

### DISCUSSION THREAD

**Discussion by jbergstroem** [community/community#5975] [Other Feature Feedback, Questions, & Ideas] [answered]
While trying out the new release-notes auto-generation feature, I entered a new tag (patch-level; 1.8.1, previous being 1.8.0) and the generator chose to diff against an older tag: v1.7.1. If there's too much complexity in the magic of figuring out what `$previous` means, perhaps allow the choice of  tag to diff against?
<img width="1019" alt="Screen Shot 2021-10-04 at 11 05 31 PM" src="https://user-images.githubusercontent.com/176984/135949061-503cdbe6-6712-416b-a2cf-29fb137d91d9.png">
<img width="1200" alt="Screen Shot 2021-10-04 at 11 05 45 PM" src="https://user-images.githubusercontent.com/176984/135949050-a7adb24a-83d1-44eb-bd83-1dbd4528b306.png">

Relevant repo: https://github.com/jbergstroem/hadolint-gh-action

Aside: not a big fan of the author being part of the changelog; but I'm probably a minority here.

**Comment by MylesBorins** [comment]
oh hey @jbergstroem been a minute! This definitely seems like a bug, I wonder if it is at all related to the v1 tag that seems to have been published on the same date as v1.8.0. Could you try deleting that tag (if it isn't being used for anyting) and let me know if that helps?

**Reply by PromoFaux** [reply]
I'm seeing this too, Just to copy what I posted in the announce thread:

---

I'm a little confused about the auto-generate release notes feature (appreciate it's still in preview)

Playing about with it to see what it does, I am not really sure how it's deciding what to put into the release notes, take for example on `pi-hole/pi-hole`, we have these tags/releases:

![image](https://user-images.githubusercontent.com/1998970/136192782-b4af613a-03b5-4fb8-9e95-dfa94fe8a3f8.png)

(Ignore the top one, it's a rolling draft)

When I add a tag of, say v5.6, and click the auto-generate button, it generates notes for contributions since `5.3.1` - Should it not be taking the most recent tag before it and going from there? (or perhaps we could have an option to manually enter the "since" tag)

![image](https://user-images.githubusercontent.com/1998970/136193107-2a0a7517-2824-4e81-95fe-e648635c8c40.png)

Now, new stuff for release is not yet in the `master` branch, rather `development`, but the same thing happens no matter which branch I choose. Would like to use this feature, as it can be quite tedious if we forget to update the rolling draft before release, so any insight would be appreciated... thanks!

**Reply by jbergstroem** [reply]
> oh hey @jbergstroem been a minute!

It has!

> This definitely seems like a bug, I wonder if it is at all related to the v1 tag that seems to have been published on the same date as v1.8.0. Could you try deleting that tag (if it isn't being used for anyting) and let me know if that helps?

I unfortunately can't. The way versions for github actions is usually by maintaining major tags in sync with latest feature/patch (another example of this pattern at `actions/checkout` here: https://github.com/actions/checkout/tags).

Happy to debug in a separate repo, can look at that later.

**Comment by PromoFaux** [comment]
I've been able to reproduce it on a play repo, too, but can't see anything obvious that would cause it. I did have a hunch that it might have been something to do with having a tag that wasn't connected to a release, after seeing one in each of mine and @jbergstroem's screenshots

https://github.com/PromoFaux/TestRepo/

![image](https://user-images.githubusercontent.com/1998970/136271061-b84d6560-a90d-4db2-9be8-1d8a259d3dd0.png)

![image](https://user-images.githubusercontent.com/1998970/136271107-6ee18590-3e1b-4a94-8240-04a31cff54e1.png)

And actually, even after publishing the `v4` release, if I draft a `v5`, I get the same thing

![image](https://user-images.githubusercontent.com/1998970/136271379-4691e078-1e15-4fa4-82ef-ff6402eedef3.png)

![image](https://user-images.githubusercontent.com/1998970/136271429-48121a25-a035-4158-80f8-6d738f2b27eb.png)

Happy to grant access to anyone who wants to debug it if needed

**Reply by MylesBorins** [reply]
we released a fix for version resolution. Can you please try again and let me know if you see improvements?

**Comment by MylesBorins** [comment]
I'm not 100% but I think that moving from using pep440 to determine previous release instead of semver might fix some of the issues we are seeing. I'm not 100% that it will fix the issue but it does seem like part of the weirdness is a mix of semver and non semver compliant tags.

In the case of the action with the v1 tag there might be even more weirdness in the UI and we might need to do some additional work to figure this out.

**Reply by jbergstroem** [reply]
> I'm not 100% that it will fix the issue but it does seem like part of the weirdness is a mix of semver and non semver compliant tags.

Trying to understand how pep440 and semver strategies would apply to mine and @PromoFaux's cases. We both seem to semver; and the git sha that the releases/tags point to are incremental date-wise. 

I'm sure I'm missing something!

**Reply by MylesBorins** [reply]
I think there might be some wrong assumptions in our existing Semver algorithm, especially what we do when we encounter a non compliant tag. Revisiting / implementing at the very least gives us a good chance to poke at what's going on and find an improvement

**Reply by jbergstroem** [reply]
Thanks for elaborating. Point me to how I can assist!

**Reply by PromoFaux** [reply]
> I think that moving from using pep440 to determine previous release instead of semver might fix some of the issues we are seeing

Please forgive a very naive question, but I'm a big fan of the ol' work smarter not harder philosophy... Is there anything stopping just taking the previous release/tag chronologically?

**Reply by MylesBorins** [reply]
@PromoFaux the last tag chronologically may be a pre-release.

**Reply by PromoFaux** [reply]
Hmm, yeah OK that makes sense I think. Is the mechanism in the background looking at tags or at Github releases? If the latter, are pre-releases not marked as such?

**Reply by MylesBorins** [reply]
As a quick update we are going to be doing a spike on the current algorithm and examine the various edge cases identified here. We hope to find a solution to fix as many of these edge cases as possible in the coming weeks

**Comment by MylesBorins** [comment]
Hey all. We just rolled out what should hopefully be a fix for many edge cases folks have identified. Can you please give release note generation another go and let us know how it works out. Thank you!

**Reply by Katsute** [reply]
It doesn't seem to be working for regular semver, when I create a draft release for <https://github.com/KatsuteDev/Chrome-Can-You-Not/releases> as `1.2.0` and it's comparing it with `1.0.0` and not `1.1.0`

**Reply by jbergstroem** [reply]
@MylesBorins my use case is fixed.

**Reply by PromoFaux** [reply]
@MylesBorins - Still not working for us on (at least):

pi-hole/AdminLTE (prev tag = `v5.8`)
![image](https://user-images.githubusercontent.com/1998970/140030356-76e0e99b-999d-451d-adf1-73f44d1c1c9a.png)

pi-hole/FTL (prev tag = `v5.11`)

![image](https://user-images.githubusercontent.com/1998970/140030577-0397991c-ae2a-4b09-922a-88a6319e3d86.png)

pi-hole/pi-hole (prev tag = `v5.6`)

![image](https://user-images.githubusercontent.com/1998970/140030656-03568b75-15e3-4601-92f3-f267d7c22f75.png)


Something I'm noting between all three here is that the auto generation _appears_ to be going back to the last tag that had a three part version number - we tend not to add the `.0` patch part on new minor versions (though we do include `.0` on a new major, interestingly... not really given it much though before)

**Reply by MylesBorins** [reply]
Thanks for sharing the examples folks. Escalating to egnineering

**Reply by jimAScott** [reply]
Hi there, 

I'm also facing the same issue. The last 6 releases I have created all try and Diff between the release I made on the 27 Oct 2021? 
I am incrementing the the following way (Major.Minor.Hotfix version) 
25.4
25.3
25.3
25.2.1
25.2
25.1
24.7
24.6.1
24.6

It would be good to have an update.

**Reply by MylesBorins** [reply]
While those version ranges are all valid pep440, they are not valid semver which confuses our current algorithm. We are exploring switching to pep440 to see if we can make a better experience for lots of folks and at the same time giving a clear warning when we were not able to accurately determine the range.

**Comment by planetf1** [comment]
I have been trying to create a release for https://github.com/odpi/egeria. We've been doing monthly released with very fixed names (.... ...2.7,2.8,2.9,2.10,2.11,3.0,3.1,3.2 etc) and I'm now doing 3.3 & see the release notes diff goes back to 2.8. I should add, each tag has a 'V' (uppercase) as a prefix. The schedule is pretty fixed and I can't see inconsistencies.

I tried this just now -- with the latest fix, and am still getting an incorrect diff

**Reply by MylesBorins** [reply]
that's for sharing, escalating to engineering

**Comment by muvaf** [comment]
Hey! We're hitting a similar issue in https://github.com/crossplane/provider-aws . I want to generate release note for `v0.21.0` which I expect to work with the diff between `v0.21.0` and `v0.20.3` but it takes `v0.13.0` for some reason. The following is the diff line at the end of the generated release notes:
```
**Full Changelog**: https://github.com/crossplane/provider-aws/compare/v0.13.0...v0.21.0
```

The latest release is `v0.20.3` and the following is the list of the latest tags. They follow a very similar pattern and we were able to get the diff logic work correct in the earlier releases. Happy to help debugging in any way.
<img width="461" alt="resim" src="https://user-images.githubusercontent.com/7584126/142277205-c7705717-681c-47f3-a256-825fbd44fea4.png">

My two cents, it'd be great if we can choose the base tag as configuration in the UI for the generation so that it doesn't have to know which tag it should be based on.

**Reply by MylesBorins** [reply]
hey @muvaf

It seems that the generated release notes are not working properly for you because of how you use release branches. Looking at your `master` branch I can see the last known tag is for commit 34522c5 which has both a v0.21.0-rc.0 tag and a v0.20.0 tag (these are arguably in conflict with each other).

The majority of your 0.20 tags are on the `release-0.20` branch and are on commits that don't exist on master. All of your pull-requests target master though (and I'm assuming that when you went to create the release you were using master as the target branch if you were creating the tag in the GitHub UI.

Our release note generation tool does not support release branches unless they are in sync with the main branch that you target PRs against. When cherrying picking commits or creating a new history the commit loses the semantic relationship with the PR that created it. Similarly if you have you tags on a different branch than the target branch we cannot accurately determine the previous release via semver and fall back to the last release that is navigable on that target branch.

**Reply by muvaf** [reply]
@MylesBorins We are using a bot [to backport](https://github.com/crossplane/provider-aws/pull/958) commits from `master` to release branches but that can't keep the commit hash as the same. Is there a way to make release branches compatible with the note generation or even having release branches are not OK for the tool?

**Reply by achimnol** [reply]
I have the exact same problem with @muvaf. Support for release branches with cherry-picked backports would be great! Backported commits usually preserve the issue number reference to the original PR in the commit message's title, so I think it should be possible to associate the related PR for them.

I'm also leaving my own git trailer like `Backported-From: main (21.09)` and `Backported-To: 21.03` to ensure future mechanical analysis or automation to distinguish cherry-picked backport commits. Maybe supporting a custom git trailer like `Associated-Pull-Request: 123` would also be a good trade-off for cases where backported commit's titles do not contain the PR number.

For reference, here are some of my repositories that use the "cherry-picked backports with release branches" policy are:
 - [lablup/backend.ai-manager](https://github.com/lablup/backend.ai-manager)
 - [lablup/backend.ai-agent](https://github.com/lablup/backend.ai-agent)
 - [lablup/backend.ai-common](https://github.com/lablup/backend.ai-common)

**Reply by cyx2** [reply]
Hi there, we have recently changed our naming strategy from ```v0.11.1``` to ```v2021-11-29.0```.  Our auto-generate release notes always refers back to the last release of the old format.  Is there any way to change this other than deleting all of the old tags?

Thank you 🙏 

@MylesBorins

**Reply by MylesBorins** [reply]
Hey @cyx2. It seems like you have moved from semver to a calendar based versioning scheme. There are a variety of projects using [calver](https://calver.org/) that you might want to explore using rather than what you are using right now. Particularly most project use `.` separators between dates rather than `-`. We are working on a change based on [pep440](https://www.python.org/dev/peps/pep-0440/) to support any arbitrary number of dot separated integers, so moving to a scheme like `v2021.11.29.0` would work after we roll out the change.

In the mean time you are indeed kind of stuck. Our algorithm right now looks for previous valid semver associated with a release, but perhaps we should short circuit that and look for the previous release in the case that the proposed new tag itself is not valid.

Thanks for reporting

**Reply by cyx2** [reply]
Very helpful response, thank you!  Is there a way I can subscribe to notifications for the pep440 change?

**Comment by pelletier** [comment]
If it helps troubleshooting it, I'm facing the same issue on https://github.com/pelletier/go-toml. I am trying to generate the changelog for a new tag `v2.0.0-beta.5` on the `v2` branch. I am expecting the generated changelog to range from the latest commit on the `v2` branch, up to and excluding the last released commit on the same branch, in that case `v2.0.0-beta.4`. Instead, the changelog is generated all the way back to `v1.8.1`, for no reason that I can tell. Note that I was able to successfully use this feature when releasing `v2.0.0-beta.4`, so it looks like a regression. 

![image](https://user-images.githubusercontent.com/172804/143176577-f14c2c2f-b0c2-4ad1-bdc9-e5cfebc81d17.png)

**Reply by MylesBorins** [reply]
Our algorithm does not look at RCs as a previous release. If we did that it wouldn't be possible to create a new release properly when ready to promote from RC

**Reply by pelletier** [reply]
Thank, good to know. Any plan on allowing to pick a "since" manually? No problem if not, I'll just look at something else to generate release notes.

**Reply by MylesBorins** [reply]
The generate notes API allows you to manually set the range. We are scoping something for the UI

**Comment by MylesBorins** [comment]
Hey all,

We have a new feature flagged version determination algorithm we'd love to test out. If you are still having issues with getting the correct previous release we'd love to flag in your repo and see if it fixes things. Please respond here with the repo you'd like us to turn it on for

**Reply by pelletier** [reply]
@MylesBorins would love to try it out on https://github.com/pelletier/go-toml.

**Reply by cyx2** [reply]
@MylesBorins we'd also be interested in trying it out on [10gen/mothra](https://github.com/10gen/mothra).  Do you have any more detailed information on how the new version determination algorithm works?

**Reply by MylesBorins** [reply]
I've added pelletier/go-toml and 10gen/mothra

The new algorithm is a riff on Pep440 which now allows for any number of `.` separated integers to make up a valid version number.

**Reply by pelletier** [reply]
@MylesBorins thanks! Still facing the same issue from https://github.com/github/feedback/discussions/5975#discussioncomment-1691576.

**Reply by cyx2** [reply]
Same for ours - still encountering the same issue.  Created a release called ```v2021.12.01.0```, but when trying to create a new release called ```v2021.12.01.1```, it still picks up ```v0.11.1``` as the base for the release notes.

**Reply by MylesBorins** [reply]
checking with engineering to figure out what's going on @cyx2 . 

@pelletier I've looked at your repo... we only support generate releases on a single release branch (e.g. all commits must be the same sha as the commit landed by the PR). We also do not support generating back to RC releases, commented more in your original post

**Reply by mourner** [reply]
@MylesBorins Can you please turn it on for https://github.com/Leaflet/Leaflet to try out? Currently the diff goes back all the way to an 8-year old tag! (v0.6.2 ... v1.8.0). I suspect this is caused by the tag not being a part of the linear history, but rather made in a "release" branch that branches out of the main branch for every release, which is a common pattern for many open source projects I participated in. 

One way to solve it would be to use semver-like sorting of tags to find the previous one as opposed to searching git history (which I suppose is what's happening now?). Ultimately the best solution would be to allow choosing the base tag manually when clicking on the button, with the automatically detected one as the first suggestion.

**Reply by MylesBorins** [reply]
@mourner I've added you in but I don't think we are going to be ablt to support your use case.

Even if we could accurately determine the correct tag we won't be able to determine the pull requests related to said release if the commits have all been cherry-picked. We utilize the semantic relationship between a commit and the PR that landed it to gather meta data to create the notes. When a commit is cherry-picked and has a new shasum that semantic relationship is lost. We've thought about ways to better support cherry picking and release branch based workflows, but these would be GitHub wide platform changes, not release specific improvemnets. (I'm one of the release managers for Node.js, so I totally empathize with wanting to simplify the release process for multiple release branches)

https://github.com/Leaflet/Leaflet/commit/bd88f73e8ddb90eb945a28bc1de9eb07f7386118

Looking at this release commit it looks like it is not associated with any branch and I'm having a hard time reconciling the history of that specific release from the rest of the main branch in the web ui (and don't have time to start cloning and poking with git right now).

It is possible that you might be able to get everything working by using our generate notes API directly and manually setting the version range. We are in the process of trying to surface this in the UI right now.

**Reply by mourner** [reply]
@MylesBorins thanks for the thorough response! Totally understand the issue with cherry-picking, but it shouldn't be a big problem in practice (any PRs that get cherry-picked in the release branch usually also land on the main branch, and we can do light editing of generated PRs afterwards anyway to fix any inconsistencies between the two).

In the case of Leaflet, the detached commits were a legacy consequence of supporting Bower many years ago (our release script created a new commit that added the built files, and then tagged that). We just got rid of this. However, it's still possible to determine the correct list of PRs between the two tags — if you go through a chain of this commit's parents, you eventually land on the main branch, and can use that as a base point.

We have a similar issue in https://github.com/mapbox/mapbox-gl-js — there's no "detached commit" problem, but we do use release branches, and currently the button calculates the list to v2.7.0 starting from v1.1.1, even though we had a ton of releases in between. I assume the same solution (looping through parents until you land on a common merge base) would work here too.

> It is possible that you might be able to get everything working by using our generate notes API directly and manually setting the version range. We are in the process of trying to surface this in the UI right now.

Yes, definitely. I couldn't derive the right query parameters in the notes request yet, but perhaps they're documented somewhere? (The UI doesn't specify the base tag on request).

**Reply by mourner** [reply]
E.g. this git command gives a correct list of PRs that landed to the main branch since the release, so it should be possible with the release-notes tool as well:

```
git rev-list --pretty=oneline v1.7.1..master
```

**Reply by cyx2** [reply]
Hi @MylesBorins - this is working well for us and we'd like to turn it on for [10gen/leafylink](https://github.com/10gen/leafylink) - is this possible

**Reply by MylesBorins** [reply]
@cyx2 added! lmk if it works

**Reply by lkiesow** [reply]
I've just tried the feature in two different repositories and both fail to recognize the previous release correctly:

- In https://github.com/opencast/pyCA creating a new `v4.5` (`v4.4` is the latest release), the algorithm recognizes `v1.0.1` as the previous release.
- In https://github.com/opencast/opencast creating `10.6` (`10.5` is the latest release), the algorithm recognizes `1.6.0` as the previous release

**Comment by MylesBorins** [comment, marked as answer]
Hey all,

We've just rolled out the new release regex to all repositories. Please take a look and see if it fixes your issues.

One of the issues folks have been running into here are repositories that are using advanced release strategies with release branches. The new regex is not going to help in those cases.

Our release generating tool is built on top of Pull-Requests and requires that your PRs target the same target branch that the release is created from.

I'm going to mark this thread as answered. Please let me know if you run into issues and we can open new discussions for additional bugs or features

**Reply by PromoFaux** [reply]
So close! It is at least not basing on 5.3.1 anymore... 

![image](https://user-images.githubusercontent.com/1998970/145871244-88aaebb7-df5e-4477-8b8f-29c917072d31.png)

https://github.com/pi-hole/pi-hole/releases

Nothing fancy about our release strategy - we simply tag `master` and away we go

**Reply by PromoFaux** [reply]
Actually, same on pi-hole/adminlte, too. 
![image](https://user-images.githubusercontent.com/1998970/145871547-91dec063-fcd7-4847-b743-dc8205b14f81.png)

And pi-hole/ftl:
![image](https://user-images.githubusercontent.com/1998970/145871668-47031b2b-bb09-4829-bb8f-17b937fc502c.png)

On each of them it appears to be skipping the last release.

pi-hole/docker-pi-hole works, though 🎉

**Reply by mourner** [reply]
Still no luck with Leaflet (picks up a range spanning years and dozens of versions), although `git rev-list --pretty=oneline v1.7.1..master` lists everything properly so unclear why it shouldn't work :(

**Reply by lkiesow** [reply]
- Works now for github.com/opencast/pyCA :tada: 
- No luck for github.com/opencast/opencast. But that's using release branches, which does make things a bit more complex. I think you could get a solution which works for that as well. E.g. picking the merge base between the latest two releases and then picking all PRs based on that but I understand if you want to keep the algorithm simple for now.

**Reply by mourner** [reply]
@MylesBorins would it be possible to expose / document the API endpoint where you could specify two commits of the main branch between which the changelog would be generated?

**Reply by MylesBorins** [reply]
https://docs.github.com/en/rest/reference/repos#generate-release-notes-content-for-a-release

**Reply by WillHalto** [reply]
👋 hey @PromoFaux ! The release note generation won't use the same commit as both the start and end of a release range - it will always look for something _before_ the current targeted commit to use as the previous version. In your https://github.com/pi-hole/pi-hole example, the current target of the new tag `v5.7` is master, whose latest commit `b5e0f14` is also referenced by tag `v5.6`. So both `v5.7` and `v5.6` are pointing to the same commit. I'd think that once any new commits are added to master, this will behave how you expect!

**Reply by PromoFaux** [reply]
Ah.. thanks @WillHalto that makes a sort of sense, I think because we are so used to just keeping a draft release as we merge into `development` branches, I was expecting to be able to generate the release notes _before_ `development` was then merged into `master`... but thanks for the explanation... I'll give it a go when next we release!

**Comment by ross-nordstrom** [comment]
@MylesBorins has the team considered a prompt for the base tag?  It seems like that would give more flexibility and avoid confusion about the default behavior?

**Comment by shoskensMagics** [comment]
We're using a year-based version strategy (v2022.6.0 > v2022.7.0) and this doesn't work at all it seems. The v2022.7.0 release does not compare to any previous release (it just generates a link to all commits). This shouldn't be too different from a v1-2-3 scheme I would think. Anything that would hinder the recognition of the base tag here?

**Comment by KhizerRehan** [comment]
Hi, @PromoFaux How to get Automated release notes from just 1 previous tag i can see that release notes are created but it is using develop branch latest tag. Which is incorrect i want to choose latest tag of master branch although the last tag was created from master branch. Any help would be appreciated.

**Comment by jimmcslim** [comment]
Where do we find this 'release regex'? I've been looking but cannot find it in Settings.

Update: if its [this](https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes) there's no mention about regular expression support.

**Comment by jimmcslim** [comment]
Raised a new issue that will hopefully get clarified https://github.com/github/feedback/discussions/12873

**Comment by planetf1** [comment]
Not working for our https://github.com/odpi/egeria releases (which we do monthly)- I end up having to edit the list manually as it insists on going back many many releases. Most commits get done on master. We do branch to a release of the form egeria-release-36 & tag as  'V3.6' format on a release - we've adopted a similar pattern for the last 1-2 years. On occasion we backport from master to the release if we find a regression in testing.

**Comment by MylesBorins** [comment]
Hey all,

we are working on a feature to allow you to select the previous tag manually. If you would like to try it out please comment in this thread with the repository you would like us to enable the preview feature on

**Reply by ross-nordstrom** [reply]
https://github.com/quickcode-ai/quickcode, assuming you can enable it on private repos?

**Reply by PromoFaux** [reply]
Very nice! Can we try it on the following please?

pi-hole/pi-hole
pi-hole/adminlte
pi-hole/ftl
pi-hole/docker-pi-hole

**Reply by gsaraf** [reply]
This would be really helpful for us as well!

Repository is https://github.com/rafttio/raftt. (Private repo, hope that is fine!)

Thanks!

**Reply by cbernet** [reply]
https://github.com/cynapps/app_rei_industry

**Reply by kevin-wise** [reply]
https://github.com/infusionsoft/account-service

**Reply by wiardvanrij** [reply]
@MylesBorins  Could you enable this for https://github.com/thanos-io/thanos/ ? currently stuck on generating a solid diff :(

**Reply by kangarruu** [reply]
👋 Hi all, I've enabled the feature for the repos requested here. Please test it out and let us know if it helps your workflow!

**Reply by gsaraf** [reply]
Thanks! Just tested and it works great! :)

Small UX issue with the Generate button now -
<img width="931" alt="Screen Shot 2022-05-03 at 16 12 36" src="https://user-images.githubusercontent.com/5310467/166459262-af0ffc30-9c48-4eef-8a66-5b5c9eb638ee.png">

**Reply by ross-nordstrom** [reply]
LGTM with tags styled `prod-abc1234`, thanks!

**Reply by edmcnealy** [reply]
Hello, is it possible for you to enable this on a private repository for me? And I am using the [REST API](https://docs.github.com/en/rest/releases/releases#create-a-release). If you are able to enable this on my repository can I still use the API to generate the notes?

**Reply by PromoFaux** [reply]
@MylesBorins  / @kangarruu - not sure if this is related to the changes here, but I've noticed on releases going "missing" on a couple of occasions recently - [a report from a user here](https://github.com/pi-hole/AdminLTE/issues/2200) - I think I've resolved it in this case by just updating the release, but I cannot work out what is making them disappear from the list of releases - there isn't really a pattern, but perhaps you have access to some special metadata that might help solve it?

**Reply by kangarruu** [reply]
@PromoFaux  Thank you for bringing this to our attention. We have received a few similar recent reports of this issue, and the team has been investigating. I have added your case to our internal issue and they're prioritizing rolling out a fix.

**Reply by planetf1** [reply]
If enabling is still an option, the two that would help us the most are
odpi/egeria (in particular)
odpi/egeria-react-ui

We do monthly releases, created at the end of each month

**Reply by swissspidy** [reply]
I'd love to try to use this on the googleforcreators/web-stories-wp repo as well

**Reply by 1st** [reply]
We wanna try this temporary solution for our private repo `b2c2/otc_api_django`

May I ask you @MylesBorins to consider also these ideas when you will improve the current solution:
- we have two parallel releases - v1 and v2 at the same time. So, we need to make a diff between the v1 and v2 releases separately when generating a release notes. Tag `v1-yyyymmdd` is for v1 release, and `v2-yyyymmdd` for the v2 release. We do daily releases actually, and both versions are prod ones
- marking two "Latest" releases in our case is more appropriate, as we have release of both v1 and v2 versions, and customers access them both as separate products
- when we create a new release for v2, we need to compare against the previous v2 latest release. When we do for v1 - we do compare against the v1 latest release

Generally it's how we use it but now we do all release notes manually because of this.

Please add myself as subscriber to the ticket related to this change, that I can keep track of when we can start use the improved solution, and I can test that it works and report back to you. Thanks.

**Reply by Sxnan** [reply]
I'd love to try the preview feature on the flink-extended/dl-on-flink repo. Thanks!

**Reply by john-u** [reply]
We'd like to try it https://github.com/SmartThingsCommunity/smartthings-cli

I suspect our project is quite the test for the auto range selection...it is a monorepo that has recently been migrated from lerna to [changesets](https://github.com/changesets/changesets) which creates tags for every package when released with `<package-name>@<version>`. Right now for us this is `@smartthings/cli@1.0.0-beta.x`. Due to the nature of monorepos, our packages are currently on different versions so I think it might be pulling from one of the packages that hasn't been released in awhile.

https://github.com/SmartThingsCommunity/smartthings-cli/releases/tag/%40smartthings%2Fcli%401.0.0-beta.8

Another thing that makes our workflow pretty bespoke is that we only want to use Github releases for the binary assets of our project and let all the other monorepo packages just get released to npm. So being able to target `@smartthings/cli@` via regex in an action would be valuable to us, but manual is better than nothing.

**Reply by SchrodingersGat** [reply]
@MylesBorins hey Myles, we would love to try out this feature on https://github.com/inventree/inventree

Please ping me if you can enable it for us to try! Cheers

**Reply by dalevfenton** [reply]
@MylesBorins we would like to try the beta on this repo https://github.com/castingnetworks/apps-monorepo

**Reply by MylesBorins** [reply]
@dalevfenton the feature is now available for all dot-com customers so you should already have it

**Reply by SampsonCrowley** [reply]
@MylesBorins is there a reason we're limited to previous release _tag_ instead of using the same option for target: a `previous_commitish` ? it's really limiting in generating notes for specific tags to _only_ be able to reference other tags in "what's changed" being able to just pass a plain commit hash would be extremely useful for workflows that have dev builds "mini releases" for QA to be able to see exactly what has changed in their image agains the current development sprint

**Reply by anthony-langford** [reply]
We'd like to try it on all our repositories. https://github.com/SecondCloset/silver

**Comment by kevin-wise** [comment]
Selecting it manually is a good workaround, but I'm confused as to why it's not finding the prior releases automatically.  I have several repos. All of them use semantic versioning beginning with a `v`. Nothing fancy at all. Yet it almost never gets the previous release correct. Would be happy to help you troubleshoot.  There are certainly no shortage of examples.

Auto-generated release notes are a great idea, but getting the right previous release is pretty key to making it useful.

**Comment by LyleDavis** [comment]
I'm slightly confused by the behaviour here, what does it select by default as the previous release? 

I was expecting it to ignore pre-releases but that doesn't seem to be the case.

We have a fairly simple versioning strategy, by year. For example, v2022.1.1 then v2022.2.1 then v2022.3.1. Whenever we create a release we mark it as "pre-release" until it has gone to production, at which point it is marked as the latest release. If we find a bug in the release, we fix it in master and then cut another, this time incrementing the final number. E.g. v2022.1.1 would be the first attempt at the first release of 2022, v2022.1.2 would be the second attempt at the first release of 2022.

Unfortunately it seems that when the second attempt at the release is cut, it uses the previous _pre-release_ as the basis for the release notes despite it not being considered the latest release.

**Reply by ross-nordstrom** [reply]
See this thread and add your repo to try their beta feature letting you pick a previous release manually when generating your release notes: https://github.com/github/feedback/discussions/5975#discussioncomment-2547153

**Reply by LyleDavis** [reply]
I considered it but that doesn't really help with the enforcement side of things really. It is very easy for people to click the big green button 😅 

I think that by default, pre-releases should be disregarded when generating release notes. I believe it does do that for semantic versioning strategies, but not when your versioning strategy is slightly different.

**Comment by ldurazo** [comment]
Bump

No way to select the last tag automatically instead of manually yet?

**Reply by fumieval** [reply]
At the moment, GitHub picks the most recent tag when "Previous tag: auto" is chosen. However, this causes confusion in our usecase because we create nightly tags; the release note should be comparing against the most recent release, not a nightly tag.

While it's not a critical issue because there's a dropdown list to pick a previous tag manually, using the latest release instead of the latest tag seems like a no-brainer improvement to me.

**Comment by pcharlan** [comment]
We're seeing "auto" make very strange choices.

What I want: choose the most recent tag whose commit is an ancestor of the current tag's commit.  (We use annotated tags.)

For bonus points, limit to semver tags, but what we're seeing is seemingly randomly skipping past 5 or so tags in history, or using something from many dozens of tags ago.

(Furthermore, once the tag is selected, the changelog doesn't even list the changes since that tag.  Sometimes it will be one change that was _from_ that chosen tag, and nothing since.  And even when selecting the tag I want from the drop-down instead of using auto, the changelog is wrong, but I get that this particular issue is about the choice of the "from" tag, not the changelog that is produced from that.

We have no `release.yml` configured in this repo.
